src/Functions/URL/cutToFirstSignificantSubdomain.cpp

```diff
@@ -43,34 +43,155 @@ using FunctionCutToFirstSignificantSubdomainWithWWWRFC = FunctionStringToString<
+    FunctionDocumentation::Arguments cutToFirstSignificantSubdomainRFC_arguments = {
+        {"url", "URL or domain string to process according to RFC 3986.", {"String"}}
+    };
+    FunctionDocumentation::ReturnedValue cutToFirstSignificantSubdomainRFC_returned_value = {
+        "Returns the part of the domain from the first significant subdomain up to the top-level domain, without 'www.', following RFC 3986.", 
+        {"String"}
+    };
+    FunctionDocumentation::Examples cutToFirstSignificantSubdomainRFC_examples = {
+    {
+        "Strips www prefix using RFC 3986 parsing", 
+        "SELECT cutToFirstSignificantSubdomainRFC('https://www.news.bbc.co.uk/')", 
+        "bbc.co.uk"
+    },
+    {
+        "Handles subdomains without www", 
+        "SELECT cutToFirstSignificantSubdomainRFC('https://docs.github.com/')", 
+        "github.com"
+    },
+    {   
+        "Processes simple domains", 
+        "SELECT cutToFirstSignificantSubdomainRFC('example.org')", 
+        "example.org"
+    }
+    };
+    FunctionDocumentation::IntroducedIn cutToFirstSignificantSubdomainRFC_introduced_in = {22, 10};
+    FunctionDocumentation::Category cutToFirstSignificantSubdomainRFC_category = FunctionDocumentation::Category::URL;
+    FunctionDocumentation cutToFirstSignificantSubdomainRFC_documentation = {cutToFirstSignificantSubdomainRFC_description, cutToFirstSignificantSubdomainRFC_syntax, cutToFirstSignificantSubdomainRFC_arguments, cutToFirstSignificantSubdomainRFC_returned_value, cutToFirstSignificantSubdomainRFC_examples, cutToFirstSignificantSubdomainRFC_introduced_in, cutToFirstSignificantSubdomainRFC_category};
+    
+    factory.registerFunction<FunctionCutToFirstSignificantSubdomainRFC>(cutToFirstSignificantSubdomainRFC_documentation);
+
+    FunctionDocumentation::Description cutToFirstSignificantSubdomainWithWWWRFC_description = R"(
+Similar to [`cutToFirstSignificantSubdomainWithWWW`](#cutToFirstSignificantSubdomainWithWWW) but follows stricter rules to be compatible with [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986).
+
+This variant performs more thorough URL parsing according to RFC 3986 standards, which may result in lower performance 
+compared to the standard version but provides more accurate handling of edge cases, while preserving the 'www.' prefix.
+    )";
+    FunctionDocumentation::Syntax cutToFirstSignificantSubdomainWithWWWRFC_syntax = "cutToFirstSignificantSubdomainWithWWWRFC(url)";
+    FunctionDocumentation::Arguments cutToFirstSignificantSubdomainWithWWWRFC_arguments = {
+         {"url", "URL or domain string to process according to RFC 3986."}
+    };
+    FunctionDocumentation::ReturnedValue cutToFirstSignificantSubdomainWithWWWRFC_returned_value = {
+        "Returns the part of the domain from the first significant subdomain up to the top-level domain, preserving 'www.', following RFC 3986.", 
```

Comment:
```suggestion
    FunctionDocumentation::ReturnedValue cutToFirstSignificantSubdomainWithWWWRFC_returned_value = {
        "Returns the part of the domain that includes top-level subdomains up to the first significant subdomain (with "www") if possible, otherwise returns an empty string", 
```